portal-kernel/src/com/liferay/portal/kernel/internal/service/persistence/TableMapperImpl.java

```diff
@@ -698,6 +712,15 @@ private boolean _deleteTableMapping(
 		return false;
 	}
 
+	private static <T extends BaseModel<T>> long _getCheckableCompanyId(T model) {
+
+		Map<String, Object> modelAttributes =
+			model.getModelAttributes();
+
+		return (long) modelAttributes.getOrDefault(
```

Comment:
I think this should be better in performance:
```
if (!model instanceof ShardedModel) {
return 0
}

return (ShardedModel)model).getCompanyId();
```

since the getModelAttributes iterates over all attributes and do a casting too inside the logic.

Do you have the getModelAttributes approach?

Reply:
Went back to the simpler `instanceof` solution.